src/Aave.sol

```diff
@@ -0,0 +1,249 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.16;
+
+import {IERC20} from "src/interfaces/IERC20.sol";
+import {IWETH} from "src/interfaces/IWETH.sol";
+import {Create2} from "src/lib/Create2.sol";
+import {SafeTransferLib} from "src/lib/SafeTransferLib.sol";
+
+// ============================
+// ======== Interfaces ========
+// ============================
+
+interface IAavePool {
+  struct ReserveConfigurationMap {
+    uint data; // This is a packed word, but we don't need this data here.
+  }
+
+  struct ReserveData {
+    ReserveConfigurationMap configuration; // stores the reserve configuration
+    uint128 liquidityIndex; // the liquidity index. Expressed in ray
+    uint128 currentLiquidityRate; // the current supply rate. Expressed in ray
+    uint128 variableBorrowIndex; // variable borrow index. Expressed in ray
+    uint128 currentVariableBorrowRate; // the current variable borrow rate. Expressed in ray
+    uint128 currentStableBorrowRate; // the current stable borrow rate. Expressed in ray
+    uint40 lastUpdateTimestamp; // timestamp of last update
+    uint16 id; // the id of the reserve. Represents the position in the list of the active reserves
+    address aTokenAddress; // aToken address
+    address stableDebtTokenAddress; // stableDebtToken address
+    address variableDebtTokenAddress; // variableDebtToken address
+    address interestRateStrategyAddress; // address of the interest rate strategy
+    uint128 accruedToTreasury; // the current treasury balance, scaled
+    uint128 unbacked; // the outstanding unbacked aTokens minted through the bridging feature
+    uint128 isolationModeTotalDebt; // the outstanding debt borrowed against this asset in isolation mode
+  }
+
+  function getReserveData(address asset)
+    external
+    view
+    returns (ReserveData memory);
+
+  function supply(
+    address asset,
+    uint amount,
+    address onBehalfOf,
+    uint16 referralCode
+  ) external;
+
+  function withdraw(address asset, uint amount, address to)
+    external
+    returns (uint);
+}
+
+// =========================
+// ======== Helpers ========
+// =========================
+
+// Amounts supplied or withdrawn are specified as a fraction of the user's
+// balance. You can pass up to 31 bytes of data to define this fraction. If zero
+// bytes are provided, the max amount is used. If one byte is provided, that
+// data is considered the numerator, and the denominator becomes 255, which is
+// the max value of a single byte. If two bytes are provided, the data is still
+// considered the numerator, but the denominator becomes 65_535, which is the
+// max value of 2 bytes. This pattern continues through 31 bytes, and this
+// method will revert if you try passing 32 bytes or more of calldata.
+// Realistically you'll never need that much anyway.
+function parseAmount(uint balance, bytes calldata data) pure returns (uint) {
```

Comment:
Nit: it would be nice if there were a way to specify slippage as well

Reply:
Sorry, slippage for which part exactly?

Reply:
If there is slippage between the amount we expect back from AAVE vs the amount parsed from the extra data

Reply:
Similar to above, aTokens are 1:1 with their underlying (it's a rebase token) so you should always get back the exact amount specified. The default router doesn't have any slippage so I don't think we need it either